Milestone 4 Risk Scores Generation Scripts

asuvarna31 opened the pull request:
1. Added a descriptive README
2. Added a build_model.py that loads data and trains the RF model
3. Added a clean requirements.txt file
4. Added a submit_data.py file that updates case on CommCare by submitting risk_scores as forms, [based on this repo](https://github.com/dimagi/submission_api_example/blob/master/submit_data.py) with very minor tweaks
5. [To Do] Add a sample_data.csv that can be used by the submit_data.py script

bderenzi commented on README.md:
```diff
@@ -0,0 +1,54 @@
+
+This repository provides example tooling for developing a risk score model, including the use of appropriate metrics for evaluation and examples of how to tune a model appropriately. The example is meant to illustrate the process as the details of any project will require some customization in approach.
+
+The primary aim of using a machine learning model in public health is to tell us where to focus human efforts or resources and maximise our impact. This contrasts with a traditional economist or epidemiologist approach, which strives to establish causality and determine the exact factors that are responsible for the outcome.
```
Optional, but could probably cut this

asuvarna31 replied:
Yep, cool to cut this - copied it from the report. 
We need a separate script that loads some data from commcare (via det) then transform that - load saved model - get risk scores - save the risk scores (probably a csv) - use submit_data.py to submit it to commcare. Is this correct ?

bderenzi commented on README.md:
```diff
@@ -0,0 +1,54 @@
+To avoid the previously mentioned issue with accuracy as a metric, data scientists often use the AUC (area under curve), which is the two-dimensional area underneath the receiver operating characteristic (ROC) curve, as an evaluation metric. The ROC curve plots true positive rate against false positive rate in order to visualize the performance of a model in different scenarios. Therefore, we use ...
+
+## Dataset Loading and Saving Models
+The dataset used for this analysis is the form submission data from CommCare applications. Each row in the dataset corresponds to a case and consists of all the historical form submission data for the particular case. In *build_model.py*, the dataset is loaded from a parquet file (you can similarly load a csv or sql file as well). 
```
We want to be framing this as a general purpose tool for anyone who uses the CommCare platform. 

```suggestion
This repository is specifically designed to expand the greater CommCare platform to use Machine Learning to generate a model to produce risk scores. The starting input data is assumed to be transactional (form-level) data. This data then needs to be aggregated down to a single row per 'case' about which we are creating the risk scores. The row will consists of all the historical form submission data for that particular case. In *build_model.py*, the dataset is loaded from a parquet file (you can similarly load a csv or sql file or pull from a database as well). 
```

(am wondering if we should modify the code to generalize the datasource a bit and make that a configuration step as opposed to hardcoded in the script)

asuvarna31 replied:
What do we mean by generalizing the data source ? The parquet/csv file format or the dataset format ?

bderenzi reviewed (changes requested):
I haven't finished reviewing. I left some comments, but also I think there might be a missing script or two? I think we want to modify (or use) the `submit_data` script to pull new data, use the saved model and then submit the risk score back to a case

asuvarna31 commented:
Hi, 
I have updated the README and added the missing scripts that were pointed out in the earlier review. Please review the final products instead of the individual commits.
